src/resources/HostedPages/HostedPages.model.ts

```diff
@@ -0,0 +1,87 @@
+import {Paginated} from '../BaseInterfaces.js';
+
+export interface ListHostedPagesParams extends Paginated {
+    /**
+     * A substring that must appear in a search interface configuration name for this configuration to appear in results.
+     */
+    filter?: string;
+
+    /**
+     * The sort direction of the results.
+     * Possible values:
+     * - `'asc'`: ascending order
+     * - `'desc'`: descending order.
+     */
+    order?: 'desc' | 'asc';
+}
+
+export interface HostedPage {
+    /**
+     * The unique identifier of the hosted page.
+     */
+    id: string;
+    /**
+     * A short descriptive name to help manage hosted pages.
+     */
+    name: string;
+    /**
+     * The HTML markup of the hosted page.
+     */
+    html: string;
+    /**
+     * JavaScript resource appended to the header of a specific hosted page in the target Coveo Cloud organization.
+     */
+    javascript?: HostedPageJavascriptFile[];
+    /**
+     * CSS resource appended in the header, pertaining to a specific hosted page in the target Coveo Cloud organization.
+     */
+    css?: HostedPageCssFile[];
+}
+
+export interface HostedPageCssFile {
+    /**
+     * The content of the header `style` tag. If this property is defined, the `url` property should not be specified.
+     */
+    inlineContent?: string;
+    /**
+     * The URL of CSS stylesheet. If this property is defined, the `inlineContent` property should not be specified.
+     */
+    url?: string;
+}
+
+export interface HostedPageJavascriptFile {
+    /**
+     * Whether the inline code should be treated as a JavaScript module. If this property is `true`, the `type` property will be set to `module` on the `script` tag.
+     */
+    isModule: boolean;
+    /**
+     * The content of the header `script` tag. If this property is defined, the `url` property should not be specified.
+     */
+    inlineContent?: string;
+    /**
+     * The URL of the JavaScript source file. If this property is defined, the `inlineContent` property should not be specified.
+     */
+    url?: string;
+}
+
+export interface HostedPageResponse extends Required<HostedPage> {
+    /**
+     * The creation timestamp. (ISO 8601)
+     */
+    created: string;
+
+    /**
+     * The creator principal.
+     */
+    createdBy: string;
+
+    /**
+     * The last update timestamp. (ISO 8601)
+     */
+    updated: string;
+
+    /**
+     * The last updated principal.
```

Comment:
Hmmm this is really making me wonder what `principal` means?
I guess it's principal in that sense? https://en.wikipedia.org/wiki/Principal_(computer_security)
Maybe it's just me who wasn't familiar with the term. Is that commonly known? If now, do you think it would be worth including a link?

Reply:
Not sure, it's copied from the other same params in the repo, perhaps the doc reviewer knows?

Reply:
Seems like this one slipped through the cracks without a doc review 🙃 
https://github.com/coveo/platform-client/pull/348
Maybe @sallainCoveo could confirm I have the right definition?
https://en.wikipedia.org/wiki/Principal_(computer_security)

Reply:
Hello!

I'm pretty sure you are right @jpmarceau. If I remember correctly, it identifies the last user that updated the search interface/ hosted page